### PR TITLE
Remove deprecated functions and classes

### PR DESCRIPTION
**Pull request by DanielNoord**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

We're getting there.

### CODE REVIEW

File: doc/whatsnew/fragments/8409.breaking [hunk 1]
```diff
@@ -0,0 +1,7 @@
+A number of old utility functions and classes have been removed.
+These are ``MapReduceMixin``, ``is_inside_lambda``, ``check_messages``,
+``is_class_subscriptable_pep585_with_postponed_evaluation_enabled``,
+``get_python_paht``, ``fix_import_path`` and ``get_global_option``.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/8409.breaking:4]:
```suggestion
``get_python_path``, ``fix_import_path`` and ``get_global_option``.
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
We need to explain what's the alternative, for example "To make a checker reduce map data simply implement get_map_data and reduce_map_data." etc. etc. (We can copy paste from the deprecation message and adapt).

**Comment by DanielNoord** [comment]
> We need to explain what's the alternative, for example "To make a checker reduce map data simply implement get_map_data and reduce_map_data." etc. etc. (We can copy paste from the deprecation message and adapt).

I was wondering about that. It feels as if the changelog can become very long. Shouldn't we just refer to the code and let the old `DeprecationWarnings` show what needs to be changed?

**Comment by Pierre-Sassoulas** [comment]
Yeah it's going to be long but it's the place where people are going to look first when something break, so it needs to explain what to do imo (also if someone is taking care of the deprecation warnings they won't need it).

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by DanielNoord** [comment]
But docs fail :(

**Review by Pierre-Sassoulas — APPROVED** [review]
I don't think we can use enumeration in the fragments, because it's already inside an enumeration. It's probably fixable but also not worth imo.

**Pull request merged**